libs/partners/snowflake/langchain_snowflake/search_retriever.py

```diff
@@ -0,0 +1,212 @@
+from typing import Any, Dict, List, Optional
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForRetrieverRun,
+    CallbackManagerForRetrieverRun,
+)
+from langchain_core.documents import Document
+from langchain_core.pydantic_v1 import Field, SecretStr, root_validator
+from langchain_core.retrievers import BaseRetriever
+from langchain_core.utils import (
+    convert_to_secret_str,
+    get_from_dict_or_env,
+    get_pydantic_field_names,
+)
+from langchain_core.utils.env import env_var_is_set
+from langchain_core.utils.utils import build_extra_kwargs
+
+# questions:
+# 1. is the filter argument to Cortex Search always of the format Dict[str, Dict[str, str]]?
```

Comment:
the answer is not necessarily. see docs here: https://docs.snowflake.com/en/user-guide/snowflake-cortex/cortex-search/query-cortex-search-service#filter-syntax